catalog/core/catalog-core-standardframework/src/test/java/ddf/catalog/impl/operations/RemoteDeleteOperationsTest.java

```diff
@@ -0,0 +1,326 @@
+
+        ArrayList<CatalogStore> stores = new ArrayList<>();
+        stores.add(catalogStoreMock);
+
+        when(opsCatStoreSupport.getCatalogStoresForRequest(any(), any())).thenReturn(stores);
+        when(catalogStoreMock.isAvailable()).thenReturn(false);
+
+        DeleteResponse resultDeleteResponse = remoteDeleteOperations.performRemoteDelete(
+                deleteRequest,
+                deleteResponse);
+        assertThat("Assert processing error occurs",
+                resultDeleteResponse.getProcessingErrors()
+                        .size() >= 1);
+
+        verify(opsCatStoreSupport).isCatalogStoreRequest(deleteRequest);
+        verify(opsCatStoreSupport).getCatalogStoresForRequest(any(), any());
+        verify(catalogStoreMock).isAvailable();
+    }
+
+    @Test
+    public void testNonEmptyStoreAvailable() throws Exception {
+
+        when(opsCatStoreSupport.isCatalogStoreRequest(deleteRequest)).thenReturn(true);
+        remoteDeleteOperations.setOpsCatStoreSupport(opsCatStoreSupport);
+
+        CatalogStore catalogStoreMock = mock(CatalogStore.class);
+
+        ArrayList<CatalogStore> stores = new ArrayList<>();
+        stores.add(catalogStoreMock);
+
+        when(opsCatStoreSupport.getCatalogStoresForRequest(any(), any())).thenReturn(stores);
+        when(catalogStoreMock.isAvailable()).thenReturn(true);
+        when(catalogStoreMock.delete(any())).thenReturn(deleteResponse);
+
+        DeleteResponse resultDeleteResponse = remoteDeleteOperations.performRemoteDelete(
+                deleteRequest,
+                deleteResponse);
+        assertThat("Assert return of identical deleteResponse",
+                resultDeleteResponse.getDeletedMetacards() == deleteResponse.getDeletedMetacards());
+
+        verify(opsCatStoreSupport).isCatalogStoreRequest(deleteRequest);
+        verify(opsCatStoreSupport).getCatalogStoresForRequest(any(), any());
+        verify(catalogStoreMock).isAvailable();
+        verify(catalogStoreMock).delete(any());
+    }
+
+    @Test
+    public void testNonEmptyStoreAvailableThrowIngestException() throws Exception {
+
+        when(opsCatStoreSupport.isCatalogStoreRequest(deleteRequest)).thenReturn(true);
+        remoteDeleteOperations.setOpsCatStoreSupport(opsCatStoreSupport);
+
+        CatalogStore catalogStoreMock = mock(CatalogStore.class);
+
+        ArrayList<CatalogStore> stores = new ArrayList<>();
+        stores.add(catalogStoreMock);
+
+        IngestException ingestException = new IngestException();
+
+        when(opsCatStoreSupport.getCatalogStoresForRequest(any(), any())).thenReturn(stores);
+        when(catalogStoreMock.isAvailable()).thenReturn(true);
+        when(catalogStoreMock.delete(any())).thenThrow(ingestException);
+
+        DeleteResponse resultDeleteResponse = remoteDeleteOperations.performRemoteDelete(
+                deleteRequest,
+                deleteResponse);
+        assertThat("Assert caught IngestException",
```

Comment:
Shouldn't this test check for processing errors on the response since one should be added if an `IngestException` is thrown?